pkg/engine/template_generator_test.go

```diff
@@ -28,3 +28,35 @@ func TestGenerateTemplateV2(t *testing.T) {
 		t.Errorf("unexpected error while running getParameterDescMap: %s", err.Error())
 	}
 }
+
+func TestGetTemplateFuncMap(t *testing.T) {
+	tg, err := InitializeTemplateGenerator(Context{})
+	if err != nil {
+		t.Errorf("unexpected error while instantiating TemplateGenerator: %s", err.Error())
+	}
+	apiModelStr := getAPIModelString()
+	cs := &api.ContainerService{}
+	err = json.Unmarshal([]byte(apiModelStr), &cs)
+	if err != nil {
+		t.Errorf("unexpected error while unmarshalling the apiModel JSON: %s", err.Error())
+	}
+	funcmap := tg.getTemplateFuncMap(cs)
+	cases := []struct {
+		key string
+	}{
```

Comment:
This is the simple "I know my pre-generated templates will depend upon this function, please validate it's there" unit test.